components/CopyToClipboard.vue

```diff
@@ -0,0 +1,50 @@
+<script setup lang="ts">
+import type { NuxtRoute, RoutesNamesList } from ".nuxt/typed-router";
+
+const props = defineProps<{
+  link?: NuxtRoute<RoutesNamesList, string, boolean>;
+  text?: string;
```

Comment:
do we ever need to copy text? Looks a bit like YAGNI (You Aint Gonna Need It)
I would simplify this component by removing unneeded features.